### ISSUE TITLE
[BUG] Saving a view of a dataset with frames

### ISSUE THREAD

**Issue by ehofesmann** [voxel51/fiftyone#870]
There is an error that gets raised when trying to save a view of a video dataset after a detection has been added to an existing frame field.
```python
import fiftyone as fo
import fiftyone.zoo as foz
from fiftyone import ViewField as F
from copy import deepcopy

dataset = foz.load_zoo_dataset("quickstart-video").clone()
sample = dataset.first()
det = deepcopy(sample.frames[1].ground_truth_detections.detections[0])
det["test"] = True
sample.frames[1].ground_truth_detections.detections.append(det)
sample.save()

dataset.filter_labels("frames.ground_truth_detections", ~F("test").exists()).save()
```

```
FieldDoesNotExist: The fields "{'frames'}" do not exist on the document "samples.2021.02.17.16.08.09"
```

**Comment by brimoor** [comment]
ok some of that code is a red herring. Here's the issue:

```py
import fiftyone as fo
import fiftyone.zoo as foz
from fiftyone import ViewField as F

dataset = foz.load_zoo_dataset("quickstart-video").clone()

dataset.filter_labels("frames.ground_truth_detections", F("label") == "vehicle").save()  # works

sample = dataset.first()
dataset.filter_labels("frames.ground_truth_detections", F("label") == "vehicle").save()  # now fails
```

The issue is reloading docs for in-memory samples. I should be able to patch this quickly.

**Comment by brimoor** [comment]
@benjaminpkane are you able to address this? I can't quite track down how frames are served on video samples. That's the issue

**Comment by benjaminpkane** [comment]
> @benjaminpkane are you able to address this? I can't quite track down how frames are served on video samples. That's the issue

Sure, one moment.

**Comment by benjaminpkane** [comment]
https://github.com/voxel51/fiftyone/commit/aa97854079ad84086a0ddab03f0e1b6a0c96e996 should fix things

I'll add a test and note. `reload()` assumes any public `attrs` are odm fields.

**Comment by brimoor** [comment]
Awesome thanks! Closing as this will be addressed in v0.7.3 release.
